main.cpp

```diff
@@ -0,0 +1,266 @@
+int serverMillis = 0;					 // time in millis since last written to server
+
+String date; // variable to store the date
+
+//Pin set up
+#define DHTPIN 4 // the pin value for the DHT11 sensor
+#define DHTTYPE DHT11
+const int MOTION_SENSOR = 15;	// PIR sensor pin
+LED *led = new LED(26, 33, 32); // the LED which shows the status of the sensors, pins 23,22,21
+const int DHT11_DELAY = 2000;	// delay for the DHT11 sensor to take readings, must be 2 seconds
+int DHT11Millis = 0;			// time in millis since DHT11 sensor took readings
+const int buttonPin = 13;
+const int STATUS_UPDATE_DELAY = 5000; // delay for the status update, must be every 5 seconds
+int statusMillis = 0;				  // time in millis since last status update
+
+//Variable initialisation
+bool firstLoop = true; //Variable to store if it is the first loop or not
+double potVal = 0;	   // Variable to store temperature value
+double humVal = 0;	   // Variable to store the humidity value
+int buttonState = 0;   //Variable for button, is not pressed when it is 0
+
+//debouncing variables
+long lastDebounceTime = 0; // the last time the output pin was toggled
+long debounceDelay = 50;   // the debounce time; increase if the output flickers
+bool snoozeBool = false;
+
+//Setting up the objects
+Thermometer *thermometer = NULL;
+Humidity *humidity = NULL;
+BUZZER *buzzer = NULL;
+DHTesp dht;					  // object to store the DHT11 sensor
+TempAndHumidity tempHum;	  // the object to store the temperature and humidity values
+TemperatureStatus tempStatus; // object to store the temperature status
+HumidityStatus humStatus;	  // object to store the humidity status
+Connect_WiFi *connect_wifi = NULL;
+PIRStatus motionSensorStatus = PIRStatus::VACANT; // object to store the PIR status
+PIR *pir = NULL;
+
+SDCard *sdcard = NULL;
+const int SD_PIN = 5; // must be pin 5
+
+/**
+ * This method returns a boolean as to whether the time that has passed is longer than the delay specified
+ */
+boolean timeDiff(unsigned long start, int specifiedDelay)
+{
+	return (millis() - start >= specifiedDelay);
+}
+
+/**
+ * This method is used at startup and initialises our sensors and pins
+ */
+void setup()
+{
+	Serial.begin(115200); // @suppress("Ambiguous problem")
+
+	pir = new PIR(MOTION_SENSOR);
+	pinMode(buttonPin, INPUT);
+	buzzer = new BUZZER(pir);
+	thermometer = new Thermometer(DHTPIN, led);
+	humidity = new Humidity(DHTPIN, led);
+
+	//OLED Screen Initialization & Setup
+	display.init();
+	display.clear();
+	display.flipScreenVertically();
+	display.setTextAlignment(TEXT_ALIGN_LEFT);
+	display.setFont(ArialMT_Plain_16);
+	display.drawString(0, 0, "WillCodeForBeer");
+	display.drawString(0, 18, "Running Setup()");
+	display.display();
+
+	dht.setup(DHTPIN, DHTesp::DHT11); // set up the DHT11 sensor
+
+	Connect_WiFi->connectToHotspot();
+	Connect_WiFi->writeToServer();
+
+	sdcard = new SDCard(SD_PIN, date); // set up SD card, must be done after wifi setup otherwise date and time wont work

```

Comment:
this will need to be the date variable from the connect wifi class (Connect_Wifi->date) as it is set in the wifi connection method